### PR TITLE
[US] As a user, I want to query the current version of the registry

### LINKED ISSUE #79: [US] As a user, I want to query the current version of the registry
As a user
I want to query the current version of the registry,
So that I can know if a certain attestation was issued under a previous version

----

**Description**

This is a simple uint16 field that auto-increments every time there has been a deployment of one of the contracts - this can be tied to a release number

We just need a state variable on one of the contracts, e.g. the attestation registry and a function to increment the version number

### PR DESCRIPTION
**Pull request by orbmis**
## What does this PR do?

As a user, I want to query the current version of the registry

This PR basically sets the version in the attestation registry, and allows the contract owner to increment it

### Related ticket

Fixes #79 

### Type of change

- [ ] Chore
- [ ] Bug fix
- [X] New feature
- [ ] Documentation update

## Check list

- [X] Unit tests for any smart contract change
- [X] Contracts and functions are documented

### CODE REVIEW

File: src/AttestationRegistry.sol [hunk 1]
```diff
@@ -101,4 +104,22 @@ contract AttestationRegistry is Initializable {
...
   }
+
+  /**
+   * @notice Increments the registry version
+   * @return The new version number
+   */
+  function incrementVersionNumber() public onlyOwner returns (uint16) {
+    version++;
+
+    return version;
```
Comment (alainncls) [inline src/AttestationRegistry.sol:115]:
```suggestion
return version++;
```
Wdyt?

File: test/AttestationRegistry.t.sol [hunk 2]
```diff
@@ -146,4 +146,18 @@ contract AttestationRegistryTest is Test {
     assertEq(attestation1.version, attestation2.version);
     assertEq(attestation1.attestationData.length, attestation2.attestationData.length);
   }
+
+  function test_incrementVersionNumber() public {
+    vm.assume(attestationRegistry.getVersionNumber() == 0);
+
+    attestationRegistry.incrementVersionNumber();
```
Comment (alainncls) [inline test/AttestationRegistry.t.sol:153]: Maybe check the returned value is `1`?

**Review by alainncls — CHANGES_REQUESTED** [review]

**Review by 0xEillo — COMMENTED** [review]
Do we want to implement semantic versioning here? so following the:

numerical components: MAJOR.MINOR.PATCH

where version would be a string concatenation

**Review by alainncls — CHANGES_REQUESTED** [review]
I've just realized `Ownable` is not "upgradeable".
Open Zeppelin is working on a v5 for their contracts, including a `OwnableUpgradeable`. In the meantime we can't use `Ownable` in a contract that will be made upgradeable (cf. #83 )

**Review by alainncls — APPROVED** [review]

**Pull request merged**